includes/class-newspack-popups-inserter.php

```diff
@@ -184,89 +199,77 @@ public static function insert_popup( $content = '', $popup = [] ) {
 	 */
-	public static function popup_access() {
+	public static function insert_popups_amp_access( $popups ) {
 		if ( is_admin() ) {
 			return;
 		}
 
-		$popup = self::popup_for_post();
+		$popups                  = self::popups_for_post();
+		$endpoint                = str_replace( 'http://', '//', get_rest_url( null, 'newspack-popups/v1/reader' ) );
+		$popups_access_providers = [];
+
+		foreach ( $popups as $popup ) {
+			// In test frequency cases (logged in site editor) and when previewing,
+			// fallback to authorization of true to avoid possible amp-access timeouts.
+			$authorization_fallback_response = (
+				( 'test' === $popup['options']['frequency'] || Newspack_Popups::previewed_popup_id() ) &&
+				is_user_logged_in() &&
+				current_user_can( 'edit_others_pages' )
+			);
 
-		if ( ! $popup ) {
-			return;
-		}
+			$amp_access_provider = array(
+				'namespace'                     => 'popup_' . $popup['id'],
+				'authorization'                 => esc_url( $endpoint ) . '?popup_id=' . esc_attr( $popup['id'] ) . '&rid=READER_ID&url=CANONICAL_URL&RANDOM',
+				'noPingback'                    => true,
+				'authorizationFallbackResponse' => array(
+					'displayPopup' => $authorization_fallback_response,
+				),
+			);
 
-		if ( ! static::assess_test_mode( $popup ) ) {
-			return;
+			array_push(
+				$popups_access_providers,
+				$amp_access_provider
+			);
 		}
 
-		// Pop-ups triggered by scroll position can only appear on Posts.
-		if ( 'scroll' === $popup['options']['trigger_type'] && ! is_single() && ! Newspack_Popups::previewed_popup_id() ) {
-			return;
+		if ( ! empty( $popups_access_providers ) ) {
+			?>
+			<script id="amp-access" type="application/json">
+				<?php echo wp_json_encode( $popups_access_providers ); ?>
+			</script>
+			<?php
 		}
-		$endpoint = str_replace( 'http://', '//', get_rest_url( null, 'newspack-popups/v1/reader' ) );
-
-		// In test frequency cases (logged in site editor) and when previewing a popup,
-		// fallback to authorization of true to avoid possible amp-access timeouts.
-		$authorization_fallback_response = (
-			( 'test' === $popup['options']['frequency'] || Newspack_Popups::previewed_popup_id() ) &&
-			is_user_logged_in() &&
-			current_user_can( 'edit_others_pages' )
-		) ? 'true' : 'false';
-		?>
-		<script id="amp-access" type="application/json">
-			{
-				"authorization": "<?php echo esc_url( $endpoint ); ?>?popup_id=<?php echo ( esc_attr( $popup['id'] ) ); ?>&rid=READER_ID&url=CANONICAL_URL&RANDOM",
-				"noPingback": true,
-				"authorizationFallbackResponse": {
-					"displayPopup": <?php echo( esc_attr( $authorization_fallback_response ) ); ?>
-				}
-			}
-		</script>
-		<?php
-		static::register_amp_scripts();
 	}
 
 	/**
 	 * Register and enqueue all required AMP scripts, if needed.
 	 */
 	public static function register_amp_scripts() {
-		if ( ! wp_script_is( 'amp-runtime', 'registered' ) ) {
+		if ( ! is_admin() && ! wp_script_is( 'amp-runtime', 'registered' ) ) {
```

Comment:
Why the `is_admin()` check here?

Reply:
Before, `register_amp_scripts` was called from `popup_access`, which [had this check](https://github.com/Automattic/newspack-popups/blob/98ddaa1c8d984045302134dfc099e1baa3ae0723/includes/class-newspack-popups-inserter.php#L223-L226). Now `register_amp_scripts` is called separately, so it has to check for itself. 
I've separated the two as part of refactoring.